_NscLib/NscContext.cpp

```diff
@@ -790,15 +791,37 @@ get_next_token:;
 								m_pStreamTop ->pszNextTokenPos++;
 							}
 							else if (c == '"' && !m_DisableNwnEeEscape)
-                            {
-                                *pszOut++ = '"';
-                                m_pStreamTop ->pszNextTokenPos++;
-                            }
+							{
+								*pszOut++ = '"';
+								m_pStreamTop ->pszNextTokenPos++;
+							}
 							else if (c == '\\' && !m_DisableNwnEeEscape)
 							{
 								*pszOut++ = '\\';
 								m_pStreamTop ->pszNextTokenPos++;
 							}
+							else if (c == 'x' && !m_DisableHexByte && strlen(m_pStreamTop ->pszNextTokenPos) >= 3) // hex value has to have length of 2
+							{
+								bool isHex = true;
+								int nValue = 0;
+								for (int pos = 1; pos <= 2; pos++)
+								{
+									char hex = m_pStreamTop ->pszNextTokenPos [pos];
+									if (isdigit(hex)) nValue = nValue * 16 + (hex - '0');
```

Comment:
Why not just use `strtol`?
Create a string on the stack, like
```const char hex[3] = { m_pStreamTop ->pszNextTokenPos[1], m_pStreamTop ->pszNextTokenPos[2], '\0'};```
And then use `strtol` to convert to a number

Reply:
Thanks, I wasn't aware of `strtol`. I'm not familiar with the stdlib.

I would still have to check both characters to fall in the range of valid hexadecimal chars though, right? In that case I don't think I would gain much by using strtol?